packages/combobox/src/ComboboxOption/ComboboxOption.tsx

```diff
@@ -62,18 +62,18 @@ export const InternalComboboxOption = React.forwardRef<
     const optionTextId = useIdAllocator({ prefix: 'combobox-option-text' });
 
     const handleOptionClick = useCallback(
-      (e: React.SyntheticEvent) => {
+      (e: React.SyntheticEvent<HTMLLIElement, Event>) => {
         // stopPropagation will not stop the keyDown event (only click)
         // since the option is never `focused`, only `aria-selected`
         // the keyDown event does not actually fire on the option element
         e.stopPropagation();
 
         if (!disabled) {
           setSelected();
-          onClick?.(value);
+          onClick?.(e);
```

Comment:
oh shoot, i think this is going to break now because i don't actually think the value is available off of the event. (i think i've tried refactoring this component before). sorry for the  FUD earlier - we might need to do something like onClick(e, value) or we can just remain consistent and not update this API. I think my comment earlier was based on a net new API not modifying an existing one.

Reply:
I think we made the ticket to add the onClick event cause consumers were asking for it. I like your suggestion tho of adding the value as well as the event.